File: algorithms/python/69.py
class Solution(object):
    def mySqrt(self, x):
        """
        :type x: int
        :rtype: int
        """
        p, q = 1, x
        while abs(p-q) > 0.001:
            p = (p+q) / 2
            q = x / p
        return int((p+q)/2)

File: algorithms/rust/69.rs
impl Solution {
    #[allow(unused)]
    pub fn my_sqrt(x: i32) -> i32 {
        let x = x as f64;
        let mut x1 = 1.0;
        let mut x2 = (x1 + x / x1) / 2.0;
        while (x1 - x2).abs() > 0.1 {
            x1 = x2;
            x2 = (x1 + x / x1) / 2.0;
        }
        return x2 as i32;
    }
}

#[allow(dead_code)]
struct Solution;

fn main() {
    println!("Hello, world!");
}
